src/common/utils.py

```diff
@@ -274,3 +274,23 @@ def read_json(file_path, default=None):
             if not isinstance(e, FileNotFoundError):
                 log.info("Unable to read file '%s' due to an exception: %s. Using default: %s", file_path, e, default)
             return default
+
+
+def validate_subprocess_argument(argument):
+    # As for https://www.gnu.org/software/bash/manual/bash.html#Lists
+    forbidden_patterns = ["&", "|",  ";",  "$",  ">", "<", "`", "\\", "!", "#", "\n"]
+
+    # Forcing the encoding to be the standard Python Unicode / UTF-8
+    # https://docs.python.org/3/howto/unicode.html
+    # https://docs.python.org/3/library/codecs.html#standard-encodings
+    _argument = (str(argument).encode("utf-8", "ignore")).decode()
+
+    if any(map(_argument.__contains__, forbidden_patterns)):
```

Comment:
I'm curious, if `_argument` is a string at this point, why use `__contains__` instead of something like this?

```
any(pattern in _argument for pattern in forbidden_patterns)
```

Dunder methods are python's internal implementation. Should we stick to using exposed APIs  to avoid any surprises?